File: backend/models/product.go
package models

import (
	"gorm.io/gorm"
)

type Product struct {
	gorm.Model
	Name        string         `json:"name"`
	Description string         `json:"description"`
	Price       float64        `json:"price"`
	Images      []ProductImage `json:"images" gorm:"foreignKey:ProductID"`
	CategoryID  uint           `json:"category_id"`
	Category    Category       `json:"category" gorm:"foreignKey:CategoryID"`
	UserID      uint           `json:"user_id"`
	User        User           `json:"user" gorm:"foreignKey:UserID"`
}

type ProductImage struct {
	gorm.Model
	URL       string `json:"url"`
	ProductID uint   `json:"product_id"`
}

File: frontend/src/types/Product.ts
import type { Category } from "./Category";
import type { ProductImage } from "./ProductImage";
import type { User } from "./User";
export interface Product {
  ID: number;
  name: string;
  description: string;
  price: number;
  images: ProductImage[];
  category_id: number;
  category?: Category;
  user_id: number;
  user?: User;
}
